Community/Tdarr_Plugin_vdka_Tiered_CPU_CRF_Based_Configurable.js

```diff
@@ -154,14 +267,93 @@ const plugin = (file, librarySettings, inputs, otherArguments) => {
+  // Used from here https://blog.frame.io/2017/03/06/calculate-video-bitrates/
+  // eslint-disable-next-line no-bitwise
+
+  const mediaInfoVideoBitrate = ~~(Number(file.mediaInfo.track[1].BitRate) / 1000);
+  const calculatedBitrate = ~~(file.file_size / (duration * 0.0075));
+  let currentBitrate;
+
+  if ((typeof mediaInfoVideoBitrate !== 'undefined') && !isNaN(mediaInfoVideoBitrate)) {
+    currentBitrate = mediaInfoVideoBitrate;
+  } else {
+    currentBitrate = calculatedBitrate;
+  }
+  const sdbitratecutoff = inputs.sdbitrate_cutoff ? inputs.sdbitrate_cutoff : 999999;
+  const hdbitratecutoff = inputs.hdbitrate_cutoff ? inputs.hdbitrate_cutoff : 999999;
+  const fullhdbitratecutoff = inputs.fullhdbitrate_cutoff ? inputs.fullhdbitrate_cutoff : 999999;
+  const uhdbitratecutoff = inputs.uhdbitrate_cutoff ? inputs.uhdbitrate_cutoff : 999999;
+
+  // set crf by resolution
+  switch (file.video_resolution) {
+    case '480p':
+    case '576p':
+      bitratecutoff = sdbitratecutoff;
+      break;
+    case '720p':
+      bitratecutoff = hdbitratecutoff;
+      break;
+    case '1080p':
+      bitratecutoff = fullhdbitratecutoff;
+      break;
+    case '4KUHD':
+      bitratecutoff = uhdbitratecutoff;
+      break;
+    default:
+      response.infoLog += 'Could for some reason not detect resolution, plugin will not proceed. \n';
+      response.processFile = false;
+      return response;
+  }
+
+  // Check if inputs.bitrate cutoff has something entered.
+  // (Entered means user actually wants something to happen, empty would disable this).
+  if (bitratecutoff !== '') {
+    // Checks if currentBitrate is below inputs.bitrate_cutoff
+    // If so then cancel plugin without touching original files.
+    if (currentBitrate <= bitratecutoff) {
+      response.processFile = false;
+      response.infoLog += `Current bitrate is ${currentBitrate} thats below set bitrate cutoff`
+      + `of ${bitratecutoff}. Nothing to do, cancelling plugin. \n`;
+      return response;
+    }
+  }
+
   // check if the file is SD and sdDisable is enabled
   // skip this plugin if so
-  if (['480p', '576p'].includes(file.video_resolution) && inputs.sdDisabled === 'true') {
+  if (['480p', '576p'].includes(file.video_resolution) && inputs.sdDisabled) {
     response.processFile = false;
     response.infoLog += '☒File is SD, not processing\n';
     return response;
   }
 
+  // check if the file is hdready and hdDisabled is enabled
+  // skip this plugin if so
+  if (['720p'].includes(file.video_resolution) && inputs.hdDisabled) {
+    response.processFile = false;
+    response.infoLog += '☒File is 720p, not processing\n';
+    return response;
+  }
+
+  // check if the file is hdready and fullhdDisabled is enabled
+  // skip this plugin if so
+  if (['1080p'].includes(file.video_resolution) && inputs.fullhdDisabled) {
+    response.processFile = false;
+    response.infoLog += '☒File is 1080p, not processing\n';
+    return response;
+  }
+
```

Comment:
Move this to before your newly added bitrate cutoff. This takes less resources to run.